content/realtime.mdx

```diff
@@ -0,0 +1,263 @@
+detailed overview of filters, read the section on [Filters](#filters). You can also specify options for the webhook. We
+current support toggling speed.
+
+The data that you will receive at the designated webhook URL will be of the format:
+
+```json JSON
+{
+  "webhook_id": "6aafe01e-b7e6-4476-a888-ee1bb570ce5f",
+  "payload": {
+    "connection_id": "2b20295c-45f2-4a32-9042-6ce3cf917536",
+    "device_id": "91618d7f-4826-44f3-a2ca-1165a6865a93",
+    "system_device_id": "vehicle1",
+    "position": {
+      "timestamp": "2024-05-04T02:38:43.000000",
+      "longitude": 144.4108704,
+      "latitude": -37.1229045,
+      "heading": 338.0,
+      "speed": { "speed_kph": 82.521 }
+    }
+  }
+}
+```
+
+### WebSocket
+
+WebSocket is a protocol enabling simultaneous two-way communication over a single TCP connection. It allows full-duplex
+interaction between a client and server with lower overhead compared to HTTP polling, facilitating real-time data
+transfer. This protocol maintains an open connection, enabling the server to send content to the client without initial
+client requests and allowing continuous message exchange.
+
+When using WebSockets for realtime data collection with PathStack, you will have to first get a connection URL using the
+`/realtime/connect` endpoint. This will return a unique URL which you can use to create a WebSocket connection with
+PathStack. See below for an example of how to connect to PathStack over websockets using that URL. Note, you will need
+to install the websockets library with `pip install websockets`.
+
+```python Python
+import asyncio
+import websockets
+
+async def connect():
+    uri = "wss://api.pathstack.io/realtime/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
+    async with websockets.connect(uri) as ws:
+        # set the filters and options for the WebSocket
+        payload = {
+            "filters": {"device_ids": ["91618d7f-4826-44f3-a2ca-1165a6865a93"]},
+            "options": {"include_speed": True},
+        }
+        await ws.send(json.dumps(payload))
+
+        while True:
+          response = await websocket.recv()
+          data = json.loads(response)
+          print(f"Received {data}")
+
+if __name__ == "__main__":
+    asyncio.run(connect())
+```
+
+The data that you will receive from the WebSocket will be of the format:
+
+```json JSON
+{
+  "connection_id": "2b20295c-45f2-4a32-9042-6ce3cf917536",
+  "device_id": "91618d7f-4826-44f3-a2ca-1165a6865a93",
+  "system_device_id": "vehicle1",
+  "position": {
+    "timestamp": "2024-05-04T02:38:43.000000",
+    "longitude": 144.4108704,
+    "latitude": -37.1229045,
+    "heading": 338.0,
+    "speed": { "speed_kph": 82.521 }
+  }
+}
+```
+
+If there are any errors, they will be received in the format:
+
+```json JSON
+{
+  "error": "<error description>"
+}
+```
```

Comment:
You should note here that WebSocket clients should detect disconnections and reconnect, and that the URL we return from `/realtime/connect` is only valid for 5 minutes, so the caller should call `/realtime/connect` again to get a new URL for reconnection.